### PR TITLE
[Google Blockly] remove modalFunctionEditor experiment

### PR DESCRIPTION
**Pull request by mikeharv**
This removes the need for an experiment flag in order for users to interact with the modal function editor in its current state. We believe this is safe to do because as of https://github.com/code-dot-org/code-dot-org/pull/54175 there are no levels for Google Blockly lab types that have the modal function editor enabled.

Technically this change involved a few things:
1. Removing the experiment and updating conditionals that referenced it
2. Removing the non-experiment flow where a new procedure definition block was created and centered on the main workspace.

I also did some light refactoring in `behaviorBlocks.js` in regards to how we create the new definition button. The code now mostly follows the same pattern used in `procedureBlocks.js` for functions. 

## Testing story

I confirmed that the buttons for creating and editing functions/behaviors now always uses the modal, even with the experiment explicitly disabled.

There are no changes or regressions observed for levels that are not using the modal function editor setting.

### CODE REVIEW

**Review by molly-moen — COMMENTED** [review]
I think it may make sense to hold this PR until [the PR to enable the behavior editor](https://github.com/code-dot-org/code-dot-org/pull/54122) is in, as it will make the `behaviorBlocks` changes fairly different.

File: apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js [hunk 1]
```diff
@@ -279,3 +262,20 @@ function simplifyBlockStateForFlyout(block) {
...
+) => {
+  const callbackKey = 'newBehaviorCallback';
+  workspace.registerButtonCallback(callbackKey, () => {
+    Blockly.functionEditor.newProcedureCallback(BLOCK_TYPES.behaviorDefinition);
+  });
+
+  return {
+    kind: 'button',
+    text: msg.createBlocklyBehavior(),
+    // TODO: Remove the alternate callback key once we're using the new function editor
```
Comment (molly-moen) [inline apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js]: should this todo still be here?

**Review by molly-moen — APPROVED** [review]
LGTM!

File: apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js (hunk 1 above)
Comment (mikeharv) [reply apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js]: Nope! Will remove.

**Pull request merged**